### PR TITLE
[All] Fix minor compilation warnings

### PR DESCRIPTION
**Pull request by damienmarchal**
______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Review by guparan — APPROVED** [review]

File: modules/SofaMiscForceField/SofaMiscForceField_test/MeshMatrixMass_test.cpp [hunk 1]
```diff
@@ -1669,17 +1668,9 @@ class MeshMatrixMass_test : public BaseTest
         EdgeSetTopologyModifier* modifier = root->getTreeObject<EdgeSetTopologyModifier>();
         ASSERT_NE(modifier, nullptr);
```
Comment (epernod) [inline modules/SofaMiscForceField/SofaMiscForceField_test/MeshMatrixMass_test.cpp:1670]: In fact all the values below are the expected one. 
I would like to keep them for when the todo is done. Maybe move the //TODO line here at beginning and comment the whole block concerned.

**Review by epernod — CHANGES_REQUESTED** [review]

File: modules/SofaMiscForceField/SofaMiscForceField_test/MeshMatrixMass_test.cpp (hunk 1 above)
Comment (damienmarchal) [reply modules/SofaMiscForceField/SofaMiscForceField_test/MeshMatrixMass_test.cpp:1670]: Done

**Comment by fredroy** [comment]
[ci-build][with-all-tests]

File: modules/SofaMiscForceField/SofaMiscForceField_test/MeshMatrixMass_test.cpp [hunk 2]
```diff
@@ -319,7 +319,8 @@ class MeshMatrixMass_test : public BaseTest
         if(mass!=nullptr){
 
             static const MassType volume = 8.0;
-            static const MassType volumeElem = volume / 8.0; // 8 hexa in the grid
+            //dmarchal, TODO(epernod): some use of the following value needs to be done
+            //static const MassType volumeElem = volume / 8.0; // 8 hexa in the grid
```
Comment (epernod) [inline modules/SofaMiscForceField/SofaMiscForceField_test/MeshMatrixMass_test.cpp]:
```suggestion
```

File: modules/SofaMiscForceField/SofaMiscForceField_test/MeshMatrixMass_test.cpp [hunk 3]
```diff
@@ -1669,17 +1670,9 @@ class MeshMatrixMass_test : public BaseTest
...
-        static const MassType expectedDensity = 1.0;
-        static const MassType expectedTotalMass = expectedDensity * volume;
-        static const MassType massElem = expectedDensity * volumeElem;
-
         const VecMass& vMasses = mass->d_vertexMass.getValue();
         const VecMass& eMasses = mass->d_edgeMass.getValue();
-        static const MassType refValueV = (MassType)(expectedDensity * volumeElem * 1 / 6);
-        static const MassType refValueE = (MassType)(expectedDensity * volumeElem * 1 / 12);
-        static const MassType wrongValue = 0; // TODO epernod 2021-06-29: MeshMatrixMass based on edge topology doesn't support topological changes
+       static const MassType wrongValue = 0; // TODO epernod 2021-06-29: MeshMatrixMass based on edge topology doesn't support topological changes
```
Comment (epernod) [inline modules/SofaMiscForceField/SofaMiscForceField_test/MeshMatrixMass_test.cpp]:
```suggestion
       static const MassType wrongValue = 0; 
       // TODO epernod 2021-06-29: MeshMatrixMass based on edge topology doesn't support topological changes
       // Keeping expected values for record
       /*
       static const MassType volume = 2.0;
       static const MassType volumeElem = volume / 3.0; // 3 edges in the grid            
       static const MassType expectedDensity = 1.0;
       static const MassType expectedTotalMass = expectedDensity * volume;
       static const MassType massElem = expectedDensity * volumeElem;
       static const MassType refValueV = (MassType)(expectedDensity * volumeElem * 1 / 6);
       static const MassType refValueE = (MassType)(expectedDensity * volumeElem * 1 / 12);
       */
```

**Review by epernod — COMMENTED** [review]
sorry previous comment was not clear. Here is my suggestion

**Review by epernod — APPROVED** [review]

**Pull request merged**